ui/src/app/base/components/NodeName/NodeName.tsx

```diff
@@ -33,13 +34,14 @@ export type FormValues = {
 
 const hostnamePattern = /^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])*$/;
```

Comment:
We should add this to `base/validation.ts` and reuse it for other node types

Reply:
Yeah good idea, we need to use it for all the node create/edit forms so I've added an issue: https://github.com/canonical-web-and-design/maas-ui/issues/3806.